index.js

```diff
@@ -1,5 +1,99 @@
-console.log("hola");
+"use strict";
+const listOfCards = document.querySelector("#lista");
+const shuffle = document.querySelector(".shuffle");
+const topCard = document.querySelector("#top-card");
 
-const mainContainer = document.querySelector("#app");
+//GENERANDO ARRAY CON CARTAS
 
-mainContainer.innerText = "Hello World";
+const cardGeneration = function () {
+  const cardsArrGen = [];
+  const clubs = ["clubs", "diamonds", "hearts", "spades"];
+  const cards = ["As", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K"];
+
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < 13; j++) {
+      const cardObj = {
+        suit: clubs[i],
+        number: cards[j],
+        color: clubs[i] === "hearts" || clubs[i] === "spades" ? "red" : "black",
+      };
+
+      switch (cardObj.suit) {
+        case "clubs":
+          cardObj.icon = "fab fa-canadian-maple-leaf";
+          break;
+        case "diamonds":
+          cardObj.icon = "fas fa-gem";
+          break;
+        case "hearts":
+          cardObj.icon = "fas fa-heart";
+          break;
+        case "spades":
+          cardObj.icon = "fas fa-candy-cane";
+          break;
+      }
+
+      cardsArrGen.push(cardObj);
+    }
+  }
+  return cardsArrGen;
+};
+
+const cardsArr = cardGeneration();
+console.log(cardsArr);
+
+//FUNCION PARA REVOLVER CARTAS
+
+const revolver = function (arr) {
+  const arrCopy = arr.slice();
```

Comment:
Good usage of `arr.slice()` it's a good shortcut to copy an array but a more transparent or easy to read way is with the spread operator like this
```suggestion
  const arrCopy = [...arr];
```